modular_skyrat/modules/assault_operatives/code/base_alarm.dm

```diff
@@ -0,0 +1,78 @@
+/obj/machinery/base_alarm
+	name = "base alarm"
+	desc = "Pull this to alert the guards!"
+	icon = 'modular_skyrat/modules/assault_operatives/icons/alarm.dmi'
+	icon_state = "alarm"
+	max_integrity = 250
+	integrity_failure = 0.4
+	use_power = NO_POWER_USE
+	resistance_flags = FIRE_PROOF
+
+	light_power = 0
+	light_range = 4
+	light_color = COLOR_VIVID_RED
+
+	//Trick to get the glowing overlay visible from a distance
+	luminosity = 1
+
+	var/alarm_playing = FALSE
+	/// Are we triggered?
+	var/triggered = FALSE
+	/// Currently connected alarms.
+	var/list/obj/machinery/base_alarm/alarms = list()
+	/// The area that we use to trigger other alarms.
+	var/area/myarea = null
+
+/obj/machinery/base_alarm/Initialize()
+	. = ..()
+	update_icon()
+	myarea = get_area(src)
+	for(var/obj/machinery/base_alarm/alarm in myarea)
+		alarms.Add(alarm)
+
+/obj/machinery/base_alarm/Destroy()
+	LAZYREMOVE(alarms, src)
+	return ..()
+
+/obj/machinery/base_alarm/attack_hand(mob/user)
+	add_fingerprint(user)
+	to_chat(user, span_notice("You trigger [src]!"))
+	playsound(src, 'sound/machines/pda_button1.ogg', 100)
+	trigger_alarm()
+
+/obj/machinery/base_alarm/attack_ai(mob/user)
+	return attack_hand(user)
+
+/obj/machinery/base_alarm/attack_robot(mob/user)
+	return attack_hand(user)
+
+/obj/machinery/base_alarm/proc/trigger_alarm()
+	if(triggered)
+		reset()
+	else
+		alarm()
+
+/obj/machinery/base_alarm/proc/alarm()
+	for(var/i in alarms)
```

Comment:
You should put the type here and use as anything, it'd be easier, if you know for sure there won't be any nulls in there. And you can always add a null check just to be safe (which I'd recommend regardless)